### PR TITLE
[MB-8340] Update react-query from 2.26.4 to 4.x

### PR DESCRIPTION
**Pull request by afoon**
## Summary

This PR updates react-query to v4. It encompasses task found in both the update to [v3](https://dp3.atlassian.net/browse/MB-8340) and [v4](https://dp3.atlassian.net/browse/MB-14638) JIRA tickets 

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the Office
2. Login as a TOO
3. App works as expected

1. Access the Office
2. Login as a TIO
3. App works as expected

1. Access the Office
2. Login as a Services Counselor
3. App works as expected

1. Access the Office
2. Login as Prime Simulator
3. App works as expected

### CODE REVIEW

**Comment by afoon** [comment]
I apologize, I thought all the test were passing again and set it as ready for review. I'm returning it back to a draft while I get the test situated again.

**Review by YanZ777 — COMMENTED** [review]
Happo is showing some errors that I think are related to these changes. It looks like some of the stories will need some mock providers like what was done for the tests here.

**Comment by afoon** [comment]
Documentation will be in a separate PR since its will be in documentation repro.

File: src/hooks/queries.js [hunk 1]
```diff
@@ -59,8 +59,8 @@ import {
 import { PAGINATION_PAGE_DEFAULT, PAGINATION_PAGE_SIZE_DEFAULT } from 'constants/queues';
 
 export const useUserQueries = () => {
-  const { data = {}, ...userQuery } = useQuery([USER, false], getLoggedInUserQueries);
-  const { isLoading, isError, isSuccess } = getQueriesStatus([userQuery]);
+  const { data = {}, ...userQuery } = useQuery([USER, false], ({ queryKey }) => getLoggedInUserQueries(...queryKey));
```
Comment (afoon) [inline src/hooks/queries.js:62]: I went to look at React Query v5 to see if there would be any breaking changes and one is reducing the number of acceptable ways to write a query or mutation to just one. 
```js
// old. syntax works in v4 but not in v5
useQuery([USER, false], ({queryKey}) => getLoggedInUserQueries(...queryKey))

// new. syntax works in v5 and in v5
useQuery({
   queryKey: [USERS, false],
   queryFn: ({queryKey}) => getLoggedInUsers(..queryKey),
  })
```
Should we handle those changes now or do when the v5 update happens?

**Comment by afoon** [comment]
@AnnaGingle I deployed the branch to Experimental for you to be able to check it out. The storyboard stories were updated to add a wrapper so they would update information correctly. This change should not be visible since this update just changed things under the hood. So testing would just making sure the Office App looks as expected.

**Review by AnnaGingle — APPROVED** [review]

File: .eslintrc.js [hunk 2]
```diff
@@ -105,9 +105,10 @@ module.exports = {
     'import/resolver': {
       node: {
         moduleDirectory: ['src', 'node_modules'],
-        extensions: ['.js', '.jsx', '.ts', '.tsx'],
+        extensions: ['.js', '.jsx'],
       },
     },
+    'import/ignore': ['.coffee$'],
```
Comment (rogeruiz) [inline .eslintrc.js:127]: As an IS3 reviewer, I have no issues with this section. It makes the frontend code more resilient to errors and removes some unused file extensions such as `ts`, `tsx`, and `coffee` from our static analysis configuration.

File: src/components/Office/RequestedShipments/SubmittedRequestedShipments.jsx [hunk 3]
```diff
@@ -90,41 +90,51 @@ const SubmittedRequestedShipments = ({
...
+        {
+          moveTaskOrderID: moveTaskOrder.id,
+          ifMatchETag: moveTaskOrder.eTag,
+          mtoApprovalServiceItemCodes,
+          normalize: false,
+        },
+        {
+          onSuccess: async () => {
+            try {
+              await Promise.all(
```
[the remark this answers is not in the document]
Comment (rogeruiz) [reply src/components/Office/RequestedShipments/SubmittedRequestedShipments.jsx:103]: 🙌🏼

File: src/components/Office/RequestedShipments/SubmittedRequestedShipments.jsx [hunk 4]
```diff
@@ -90,41 +90,51 @@ const SubmittedRequestedShipments = ({
...
+                  }
+                  return approveMTOShipment(
+                    {
+                      shipmentID: shipment.id,
+                      operationPath,
+                      ifMatchETag: shipment.eTag,
+                      normalize: false,
+                    },
+                    {
+                      onError: () => {
```
[the remark this answers is not in the document]
Comment (rogeruiz) [reply src/components/Office/RequestedShipments/SubmittedRequestedShipments.jsx:118]: Love the references here. Thanks for this comment and the others similar to these.

File: src/components/Office/ShipmentForm/ShipmentForm.jsx [hunk 5]
```diff
@@ -97,6 +100,12 @@ const ShipmentForm = (props) => {
     },
   });
 
+  const { mutate: mutateMoveCloseoutOffice } = useMutation(updateMoveCloseoutOffice, {
```
[the remark this answers is not in the document]
Comment (rogeruiz) [reply src/components/Office/ShipmentForm/ShipmentForm.jsx:103]: Thanks for this call out.

File: src/components/Office/ShipmentForm/ShipmentForm.jsx [hunk 6]
```diff
@@ -698,6 +755,7 @@ ShipmentForm.defaultProps = {
   isForServicesCounseling: false,
   match: { isExact: false, params: { moveCode: '', shipmentId: '' } },
   history: { push: () => {} },
+  onUpdate: () => {},
```
[the remark this answers is not in the document]
Comment (rogeruiz) [reply src/components/Office/ShipmentForm/ShipmentForm.jsx:758]: I don't know if this link is supposed to go to Stack Overflow? I was expecting it to be linking to where the second callback is in our codebase.

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

Great work on this @afoon and @transcom/truss-pamplemoose. This was a tricky upgrade, but the project is much better for it.

File: .eslintrc.js (hunk 2 above)
Comment (rogeruiz) [inline .eslintrc.js:127]: As an IS3 reviewer, I have no issues with this section. It makes the frontend code more resilient to errors and removes some unused file extensions such as `ts`, `tsx`, and `coffee` from our static analysis configuration.

Comment (rogeruiz) [reply .eslintrc.js:127]: Not sure why the GH interface isn't showing the full selection, but this section of configuration also removes TS/TSX extensions and modifies warnings to errors.

File: src/components/Office/ShipmentForm/ShipmentForm.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (afoon) [reply src/components/Office/ShipmentForm/ShipmentForm.jsx:758]: Yes that is the wrong link. I've updated it. Here is the correct one: https://github.com/transcom/mymove/blob/main/src/pages/Office/ServicesCounselingEditShipmentDetails/ServicesCounselingEditShipmentDetails.jsx#L61

**Comment by mmayo-truss** [comment]
I have gone through multiple workflows for the customer, each of the office roles, and the prime UI and verified that things are working as expected.

**Review by mmayo-truss — APPROVED** [review]
LGTM. Good work @afoon this was a big undertaking.

**Pull request merged**